androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/MapsFragment.kt

```diff
@@ -36,8 +45,67 @@ class MapsFragment : Fragment() {
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
+        setupUI()
+        viewModel.state.observe(viewLifecycleOwner, ::onStateChanged)
+    }
+
+    @SuppressLint("MissingPermission")
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        val permissionResultParser = PermissionResultParser()
+        if (permissionResultParser.isFineLocationPermissionsGranted(permissions, grantResults)) {
+            viewModel.startUpdatingLocation()
+            map.isMyLocationEnabled = true
+        } else {
+            showToast("Location permissions are required to find meals")
+        }
+    }
+
+    private fun setupUI() {
         val mapFragment = childFragmentManager.findFragmentById(R.id.map) as SupportMapFragment?
         mapFragment?.getMapAsync(callback)
     }
+
+    private fun onStateChanged(state: MapsViewModel.State) {
+        when (state) {
+            MapsViewModel.State.LoadingMeals ->
+                Log.i("Map view", "Loading meals")
```

Comment:
Not sure whether we need to handle loading state on the map.  Whether we want to show loading spinner or some other UI?

Reply:
I don't think there's usually a loading state for maps, unless you want to have a refresh. The way I've handled it previously is having the map setup as an async for getting the data in the background and once that's been fetched to add the markers onto the map, so that the user can still interact with the map like scroll or zoom while the data is being fetched. I've not personally added a spinner to a map, I think that's better for a list view. There are probably better ways to do so, though :)